providers/google/src/airflow/providers/google/cloud/operators/dataproc.py

```diff
@@ -2708,6 +2710,25 @@ def _inject_openlineage_properties_into_dataproc_batch(self, context: Context) -
                 exc_info=e,
             )
 
+    def __update_batch_labels(self):
+        dag_id = re.sub(r"[.\s]", "_", self.dag_id.lower())
+        task_id = re.sub(r"[.\s]", "_", self.task_id.lower())
```

Comment:
Is there a reason not to label with `dag_display_name` ?

Reply:
Google cloud labels have their own validation so I wanted to have an explicit code that sanitizes the values. Othewise, if someone changes dag_display_name value workloads will start to fail with validation error.

Reply:
Good point.

Reply:
But users can also change dag_id
Some patterns of dag authoring migrated to use dag_display_name and hide the option to set dag_id directly.
Unless there is a strong reason to avoid I would rather we allow to set it, provide to users the needed knowledge and let them to make thier choices.

Regardless if my suggestion is not accepted we should at least explain in a user facing doc why we don't track the display name.

Reply:
I am not sure if `dag_display_name` is a better choice for identifying dags - I do not see any enforcement for it to be unique.  (Updated PR description).

We can just add one more label with display name and make it clear that it is not id, but to be honest I do not see use-case for that. What do you think?

Reply:
Didnt meant to replace dag_id.
My intention was to add another label

Reply:
But then it will be different dag, so labelling it with sanitised dag_id makes perfect sense IMHO

Reply:
Updated with another tag (label).